### PR TITLE
Minimum Qty Allowed in Shopping Cart not working on related product

### PR DESCRIPTION
**Pull request by mageprince**
### Description (*)

Solve **Minimum Qty Allowed in Shopping Cart** not working on related products

### Manual testing scenarios (*)

1. Let's take two simple products **Product A** and **Product B**
2. Set Minimum Qty Allowed in Shopping Cart = 10 in **Product A** from
Advanced Inventor->Minimum Qty Allowed in Shopping Cart
3. Assign **Product A** as a related product in **Product B**
3. Open PDP page of **Product B**
4. Check Product A checkbox from Related Products block on PDP
5. Add to cart **Product B** 

### Expected result (*)

Add to cart **Product A** with Qty = 10

### Actual result (*)

1. Add only 1 qty of Product A to the cart
2. Showing error "We don't have as many of some products as you want."

### Contribution checklist (*)
 - [ ] Pull request has a meaningful description of its purpose
 - [ ] All commits are accompanied by meaningful commit messages
 - [ ] All new or changed code is covered with unit/integration tests (if applicable)
 - [ ] All automated tests passed successfully (all builds on Travis CI are green)

### CODE REVIEW

**Comment by magento-engcom-team** [comment]
Hi @mageprince. Thank you for your contribution 
Here is some useful tips how you can test your changes using Magento test environment.
Add the comment under your pull request to deploy test or vanilla Magento instance:
- `@magento-engcom-team give me test instance` - deploy test instance based on PR changes
- `@magento-engcom-team give me 2.3-develop instance` - deploy vanilla Magento instance

For more details, please, review the [Magento Contributor Assistant](https://devdocs.magento.com/guides/v2.2/contributor-guide/contributing.html#contributor-assist) documentation

**Comment by mageprince** [comment]
@magento-engcom-team give me 2.3-develop instance

**Comment by magento-engcom-team** [comment]
Hi @mageprince. Thank you for your request. I'm working on Magento 2.3-develop instance for you

**Comment by magento-engcom-team** [comment]
Hi @mageprince, here is your [Magento instance](https://i-19566-2-3-develop.instances.magento-community.engineering).
Admin access: [https://i-19566-2-3-develop.instances.magento-community.engineering/admin](https://i-19566-2-3-develop.instances.magento-community.engineering/admin)
Login: `admin` Password: `123123q`
Instance will be terminated in up to 3 hours.

**Comment by mageprince** [comment]
@magento-engcom-team Issue reproduced in the test instance

File: app/code/Magento/Checkout/Model/Cart.php [hunk 1]
```diff
@@ -438,8 +438,14 @@ public function addProductsByIds($productIds)
                 }
                 $product = $this->_getProduct($productId);
                 if ($product->getId() && $product->isVisibleInCatalog()) {
+                    $request = null;
+                    $stockItem = $this->stockRegistry->getStockItem($productId, $product->getStore()->getWebsiteId());
+                    $minimumQty = $stockItem->getMinSaleQty();
+                    if ($minimumQty && $minimumQty > 0) {
+                        $request = $minimumQty;
```
Comment (orlangur) [inline app/code/Magento/Checkout/Model/Cart.php]: Complex logic related to inventory must not be added in `Checkout` module. We can have much more checks than MinSaleQty.

**Review by orlangur — DISMISSED** [review]

File: app/code/Magento/Checkout/Model/Cart.php (hunk 1 above)
Comment (mageprince) [reply app/code/Magento/Checkout/Model/Cart.php]: @orlangur Are you check the `addProduct()` function of `\Magento\Checkout\Model\Cart` It also check the `MinSaleQty` with same logic. I took the reference from there. You can check here: [Cart.php](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Checkout/Model/Cart.php#L372-L380)

Comment (orlangur) [reply app/code/Magento/Checkout/Model/Cart.php]: @mageprince does not look good as well, it should be better implemented as a plugin, probably legacy code.

Why do we need to duplicate this logic out of `addProduct` method?

Comment (mageprince) [reply app/code/Magento/Checkout/Model/Cart.php]: I think we misunderstood functions between addProduct() of `\Magento\Quote\Model\Quote` and `addProduct()` of `\Magento\Checkout\Model\Cart`.

**`addProduct()`** of `\Magento\Checkout\Model\Cart` **check the `minimumQty` of product** to addtocart
While **`addProductsByIds()`** of `\Magento\Checkout\Model\Cart` **not check the minimumQty** but it must required to solve this issue.

Comment (mageprince) [reply app/code/Magento/Checkout/Model/Cart.php]: @orlangur any update for this issue?

Comment (sivaschenko) [reply app/code/Magento/Checkout/Model/Cart.php]: @mageprince considering there is very similar check in [Cart.php](https://github.com/magento/magento2/blob/2.3-develop/app/code/Magento/Checkout/Model/Cart.php#L372-L380) I think it's better at least to move this check to private method and reuse in both functions. I think @orlangur would like this check to be moved to `CatalogInventory` module

Comment (mageprince) [reply app/code/Magento/Checkout/Model/Cart.php]: @orlangur @sivaschenko 
I updated the pull request as per your suggestion.
Please review it.

Comment (mageprince) [reply app/code/Magento/Checkout/Model/Cart.php]: @orlangur @sivaschenko any update on this pull request.

**Review by sivaschenko — APPROVED** [review]

**Comment by magento-engcom-team** [comment]
Hi @sivaschenko, thank you for the review.
[ENGCOM-3892](https://jira.corp.magento.com/browse/ENGCOM-3892) has been created to process this Pull Request

**Comment by magento-engcom-team** [comment]
@mageprince thank you for contributing. Please accept Community Contributors team invitation [here](https://github.com/orgs/magento/invitation) to gain extended permissions for this repository.

File: app/code/Magento/Checkout/Model/Cart.php (hunk 1 above)
Comment (mageprince) [reply app/code/Magento/Checkout/Model/Cart.php]: @orlangur can you please review the changes.

**Pull request merged**